src/lib/build.js

```diff
@@ -42,6 +45,18 @@ const getBuildOptions = ({ cachedConfig, options: { context, cwd, debug, dry, js
  */
 const runBuild = async (options) => {
   const { default: build } = await netlifyBuildPromise
+
+  // If netlify NETLIFY_API_URL is set we need to pass this information to @netlify/build
+  // TODO don't use testOpts, but add real properties to do this.
+  if (process.env.NETLIFY_API_URL) {
+    const apiUrl = new URL(process.env.NETLIFY_API_URL)
+    const testOpts = {
+      scheme: apiUrl.protocol.slice(0, -1),
+      host: apiUrl.host,
+    }
+    options = { ...options, testOpts }
+  }
```

Comment:
This is not nice,.
The `host` could be supplied by [setting](https://github.com/netlify/build/blob/1337b1b3d43bc1dce7d1e82a10159cfcd4f72c15/packages/build/src/core/main.js#L160) `apiHost` but the scheme cannot be changed besides the testOpts.

I'm also not sure if `NETLIFY_API_URL` is only for tests  or if that is meant to be used by users?

Should I add `apiScheme` to `@netlify/build`? Or is it okay this way?